applications/crossbar/src/modules/cb_allotments.erl

```diff
@@ -0,0 +1,258 @@
+resource_exists() -> 'true'.
+resource_exists(?CONSUMED) -> 'true'.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% This function determines if the parameters and content are correct
+%% for this request
+%%
+%% Failure here returns 400
+%% @end
+%%--------------------------------------------------------------------
+-spec validate(cb_context:context()) -> cb_context:context().
+validate(Context) ->
+    validate_allotments(Context, cb_context:req_verb(Context)).
+
+-spec validate(cb_context:context(), path_token()) -> cb_context:context().
+validate(Context, ?CONSUMED) ->
+    validate_consumed(Context, cb_context:req_verb(Context)).
+
+-spec validate_allotments(cb_context:context(), http_method()) -> cb_context:context().
+validate_allotments(Context, ?HTTP_GET) ->
+    load_allotments(Context);
+validate_allotments(Context, ?HTTP_POST) ->
+    cb_context:validate_request_data(<<"allotments">>, Context, fun on_successful_validation/1).
+
+-spec validate_consumed(cb_context:context(), http_method()) -> cb_context:context().
+validate_consumed(Context, ?HTTP_GET) ->
+    load_consumed(Context).
+
+-spec post(cb_context:context()) -> cb_context:context().
+post(Context) ->
+    update_allotments(Context).
+
+%%%===================================================================
+%%% Internal functions
+%%%===================================================================
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Load a document from the database
+%% @end
+%%--------------------------------------------------------------------
+-spec load_allotments(cb_context:context()) -> cb_context:context().
+load_allotments(Context) ->
+    Context1 = maybe_handle_load_failure(crossbar_doc:load(?PVT_TYPE, Context)),
+    Allotments = wh_json:get_json_value(?PVT_ALLOTMENTS, cb_context:doc(Context1), wh_json:new()),
+    cb_context:set_resp_data(Context1, Allotments).
+
+
+-spec load_consumed(cb_context:context()) -> cb_context:context().
+load_consumed(Context) ->
+    Context1 = crossbar_doc:load(?PVT_TYPE, Context),
+    Allotments = wh_json:get_json_value(?PVT_ALLOTMENTS, cb_context:doc(Context1), wh_json:new()),
+    MoDb = cb_context:account_modb(Context),
+    load_consumed(Context, MoDb, Allotments).
+
+-spec load_consumed(cb_context:context(), api_binary(), wh_json:objects()) -> cb_context:context().
+load_consumed(Context, MoDb, Allotments) ->
+    {_, Result} = wh_json:foldl(fun foldl_consumed/3, {MoDb, []}, Allotments),
+    cb_context:setters(Context
+                       ,[{fun cb_context:set_resp_status/2, 'success'}
+                         ,{fun cb_context:set_resp_data/2, Result}
+                        ]).
+
+-spec foldl_consumed(api_binary(), wh_json:object(), {cb_context:context(), api_binary(), wh_json:objects()}) -> wh_json:objects().
+foldl_consumed(Classification, Value, {MoDb, Acc}) ->
```

Comment:
@Dinkor spec seems wrong. declares tuple/3 in 3rd argument but function handles tuple/2